data-access-services/accounts-service/src/main/java/at/deloittedigital/core_banking/accounts/entity/Account.java

```diff
@@ -0,0 +1,48 @@
+package at.deloittedigital.core_banking.accounts.entity;
+
+import at.deloittedigital.core_banking.lib.entity.BaseEntity;
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.ManyToOne;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+public class Account extends BaseEntity {
+
+    /**
+     * Customer to which the account belongs.
+     */
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Client client;
+
+    /**
+     * Frequency at which statements are sent to the customer, e.g. monthly, weekly or per
+     * transaction.
+     */
+    private String frequency;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Account)) {
+            return false;
+        }
+        Account account = (Account) o;
+        return Objects.equals(getId(), account.getId());
+    }
+
+    @Override
+    public int hashCode() {
+        return 0;
```

Comment:
Shouldn't `hashCode()` use the same field (id) as `equals()` for the sake of consistency? Also, with `0` everything will land in the same bucket if we were to use it in a map.

Reply:
I usually refer to these articles about that topic:

- https://vladmihalcea.com/the-best-way-to-map-a-onetomany-association-with-jpa-and-hibernate/ and 
- https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/

We should keep them consistent because we rely on equality for the `removeAccount` method. E.g. otherwise objects would differ before and after persisting.

Reply:
👍